src/VisualStudio/Core/Def/Experimentation/EnhancedColorExperiment.cs

```diff
@@ -0,0 +1,354 @@
+                        IsDefaultForeground(colorItemMap, ClassificationTypeNames.OperatorOverloaded, LightThemeOperator) &&
+                        (IsDefaultForeground(colorItemMap, ClassificationTypeNames.ControlKeyword, LightThemePlainText) ||
+                        IsDefaultForeground(colorItemMap, ClassificationTypeNames.ControlKeyword, LightThemeKeyword));
+                }
+
+                return areColorsDefaulted;
+            }
+
+            private bool IsDefaultForeground(Dictionary<string, ColorableItems> colorItemMap, string classification, uint themeColor)
+            {
+                var color = colorItemMap[classification].Foreground;
+
+                // The color value isn't always consistent. Sometimes it'll come back as DefaultColor
+                // other times it'll come back as the theme color. Just covering all the bases.
+                return color == DefaultForegroundColor ||
+                    color == AutomaticForegroundColor ||
+                    color == themeColor;
+            }
+
+            private bool AreColorsEnhanced(Dictionary<string, ColorableItems> colorItemMap, Guid themeId)
+            {
+                bool areColorsEnhanced;
+
+                if (themeId == KnownColorThemes.Dark)
+                {
+                    // Dark Theme
+                    areColorsEnhanced = colorItemMap[ClassificationTypeNames.LocalName].Foreground == DarkThemeLocalBlue &&
+                        colorItemMap[ClassificationTypeNames.ParameterName].Foreground == DarkThemeLocalBlue &&
+                        colorItemMap[ClassificationTypeNames.MethodName].Foreground == DarkThemeMethodYellow &&
+                        colorItemMap[ClassificationTypeNames.ExtensionMethodName].Foreground == DarkThemeMethodYellow &&
+                        colorItemMap[ClassificationTypeNames.OperatorOverloaded].Foreground == DarkThemeMethodYellow &&
+                        colorItemMap[ClassificationTypeNames.ControlKeyword].Foreground == DarkThemeControlKeywordPurple &&
+                        colorItemMap[ClassificationTypeNames.StructName].Foreground == DarkThemeStructMint;
+                }
+                else
+                {
+                    // Light or Blue themes
+                    areColorsEnhanced = colorItemMap[ClassificationTypeNames.LocalName].Foreground == LightThemeLocalBlue &&
+                        colorItemMap[ClassificationTypeNames.ParameterName].Foreground == LightThemeLocalBlue &&
+                        colorItemMap[ClassificationTypeNames.MethodName].Foreground == LightThemeMethodYellow &&
+                        colorItemMap[ClassificationTypeNames.ExtensionMethodName].Foreground == LightThemeMethodYellow &&
+                        colorItemMap[ClassificationTypeNames.OperatorOverloaded].Foreground == LightThemeMethodYellow &&
+                        colorItemMap[ClassificationTypeNames.ControlKeyword].Foreground == LightThemeControlKeywordPurple;
```

Comment:
so if the user changes a single color, it's no longer enhanced?

Reply:
Well they aren't using only our enhanced colors. The way I was viewing it is that a users owns their colors once they make a change. I am hesitant to overwrite their changes. After disabling the experiment in the preview options, the user could click 'Use Defaults' from Fonts and Colors to revert their changes and our changes.

Reply:
Thanks.  That makes more sense.  Possibly worth a comment.

Reply:
There won't even be an option to use enhanced colors in case they want to opt in?

Reply:
@Neme12 All of these classifications will show up in the Fonts and Colors dialog regardless of this experimental flag. For this experiment we have a set of colors we will try to set automatically, if the user has not made their own changes to these particular classifications. If the user wishes to opt-in to the experiment and have their own colors for these classifications, they will need to reset the individual classifications to 'Default color' or they can set their entire theme using the 'Use Defaults' button in Fonts and Colors.